### PR TITLE
add TemperatureControl and tests

### PR DESCRIPTION
**Pull request by eucalvo**
I believe this is ready for a merge.

### CODE REVIEW

File: test/TemperatureControlTest.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#include <ci/ObservableDataStream.h>
+
+#include "TankControllerLib.h"
+#include "TemperatureControl.h"
+
+/*
+bundle exec arduino_ci.rb --skip-examples-compilation --testfile-select=TemperatureControlTest.cpp
+*/
+
+//  void Chiller::updateControl(double currentTemperature);   // void TemperatureControl::turnOn(bool newState) {
```
Comment (prestoncarman) [inline test/TemperatureControlTest.cpp]: Don't include commented out code.

File: test/TemperatureControlTest.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#include <Arduino.h>
+#include <ArduinoUnitTests.h>
+#include <ci/ObservableDataStream.h>
+
+#include "TankControllerLib.h"
+#include "TemperatureControl.h"
+
+/*
+bundle exec arduino_ci.rb --skip-examples-compilation --testfile-select=TemperatureControlTest.cpp
```
Comment (prestoncarman) [inline test/TemperatureControlTest.cpp]: Is this needed? It seems like a common command that a developer should know. Would it make more sense to have a section of the read me explaining how to run a single test file?

File: src/Devices/TemperatureControl.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#include <Arduino.h>
+
+// class temp control -temp setpoint +get/set temp target +virtual updateCtrl(),  subclass heater (respective code) and
```
Comment (prestoncarman) [inline src/Devices/TemperatureControl.h]: Please make comment a little more readable and put it in this format:
```
/**
 * My comment line 1
 * line 2
 */
```

**Review by prestoncarman — CHANGES_REQUESTED** [review]
Just a few comments on comments.

File: test/TemperatureControlTest.cpp (hunk 2 above)
Comment (jgfoster) [reply test/TemperatureControlTest.cpp]: The more things I don't have to remember the better; I can apply my scarce and failing memory to other things. This line shows how to run a single test and that's something that I know can be done but I don't remember how to do it. I'd rather not have to go to a README.md to piece this command line together from a page of instructions and options and arguments every time I need it.

File: test/TemperatureControlTest.cpp (hunk 1 above)
Comment (jgfoster) [reply test/TemperatureControlTest.cpp]: I agree on removing (or explaining) this.

File: src/Devices/TemperatureControl.h (hunk 3 above)
Comment (jgfoster) [reply src/Devices/TemperatureControl.h]: This does look like notes that may have been helpful during development but should be removed or expanded.

**Review by jgfoster — CHANGES_REQUESTED** [review]
A couple changes to comments.

File: test/TemperatureControlTest.cpp (hunk 2 above)
Comment (prestoncarman) [reply test/TemperatureControlTest.cpp]: Ok, can you add a comment above to make it clear what this line does and a note its a shortcut for developers.

**Review by prestoncarman — APPROVED** [review]
LGTM

**Review by jgfoster — APPROVED** [review]
LGTM

**Comment by prestoncarman** [comment]
@all-contributors please add @eucalvo for code

**Pull request merged**